src/main/java/me/arminb/sara/dao/UserDAO.java

```diff
@@ -0,0 +1,13 @@
+package me.arminb.sara.dao;
+
+import me.arminb.sara.entities.User;
+import java.util.List;
+
+public interface UserDAO {
+    public List<User> findAll();
+    public User find(String id);
```

Comment:
I would name this findById. and would add another method find(User user) which will contain a partial set of a user information that can be matched against the database. Then, it is possible to use this method for general searches like email=a@b.c&name=armin.